platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -0,0 +1,209 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.solr.factory.impl;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.response.CollectionAdminResponse;
+import org.apache.zookeeper.KeeperException;
+import org.codice.solr.factory.SolrClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+
+public class SolrCloudClientFactory implements SolrClientFactory {
+
+    protected static final Logger LOGGER = LoggerFactory.getLogger(SolrCloudClientFactory.class);
+
+    public static final String DEFAULT_ZOOKEEPER_HOST = "localhost:9983";
+
+    private static final int SHARD_COUNT = Integer.parseInt(System.getProperty("solr.cloud.shardCount", "2"));
+
+    private static final int REPLICATION_FACTOR = Integer.parseInt(System.getProperty("solr.cloud.replicationFactor", "2"));
+
+    private static final int MAXIMUM_SHARDS_PER_NODE = Integer.parseInt(System.getProperty("solr.cloud.maxShardPerNode", "2"));
+
+    private static final String THREAD_POOL_DEFAULT_SIZE = "128";
+
+    private static final ScheduledExecutorService EXECUTOR_SERVICE = createExecutorService();
+
+    private static ScheduledExecutorService createExecutorService() throws NumberFormatException {
+        Integer threadPoolSize = Integer.parseInt(System.getProperty(
+                "org.codice.ddf.system.threadPoolSize",
+                THREAD_POOL_DEFAULT_SIZE));
+        return Executors.newScheduledThreadPool(threadPoolSize);
+    }
+
+    @Override
+    public Future<SolrClient> newClient(String core) {
+        String zookeeperHosts = System.getProperty("solr.cloud.zookeeper", DEFAULT_ZOOKEEPER_HOST);
+        return getClient(zookeeperHosts, core);
+    }
+
+    public static Future<SolrClient> getClient(String zookeeperHosts, String collection) {
+        RetryPolicy retryPolicy = new RetryPolicy()
+                .retryWhen(null)
+                .withBackoff(10,
+                TimeUnit.MINUTES.toMillis(1),
```

Comment:
It was when I first wrote it.  Looks like the latest formatter handles it differently now.